### PR TITLE
fix: read fstab from code bus

### LINKED ISSUE #18: read fstab via code-bus bucket to avoid github authentication
private github repositories can't be accessed so it's better to read the fstab from the code-bus bucket.

### PR DESCRIPTION
**Pull request by dominique-pfister**
fix #18

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -57,7 +59,28 @@ async function main(req, context) {
...
+      AWS_S3_REGION,
+      AWS_S3_ACCESS_KEY_ID,
+      AWS_S3_SECRET_ACCESS_KEY,
+      bucket: 'helix-code-bus',
+      readOnly: true,
+      log,
+    });
+    const res = await codeStorage.load(`${owner}/${repo}/${ref}/fstab.yaml`);
+    fstab = await new MountConfig().withSource(await res.text()).init();
+  } finally {
```
Comment (tripodsan) [inline src/index.js:76]: maybe log an error if the fstab could not be loaded?

File: src/storage.js [hunk 2]
```diff
@@ -138,19 +138,38 @@ class AWSStorage {
...
+    const { log } = this;
+
+    const input = {
+      Bucket: this.bucket,
+      Key: key,
+    };
+
+    const result = await this.client.send(new GetObjectCommand(input));
+    log.info(`Object downloaded from: ${this.bucket}/${key}`);
+    return new Response(result.Body, {});
```
Comment (tripodsan) [inline src/storage.js:162]: why create a response and not just return the body ?

**Review by tripodsan — APPROVED** [review]

File: src/index.js (hunk 1 above)
Comment (dominique-pfister) [reply src/index.js:81]: Added code for testing response from `load`

File: src/index.js [hunk 3]
```diff
@@ -57,7 +59,33 @@ async function main(req, context) {
...
+      AWS_S3_SECRET_ACCESS_KEY,
+      bucket: 'helix-code-bus',
+      readOnly: true,
+      log,
+    });
+    const buffer = await codeStorage.load(`${owner}/${repo}/${ref}/fstab.yaml`);
+    if (!buffer) {
+      return new Response(`${owner}/${repo}/${ref}/fstab.yaml not found in bucket 'helix-code-bus'`, {
+        status: 400,
+      });
```
Comment (tripodsan) [inline src/index.js:78]:
```suggestion
      return new Response(``, {
        status: 400,
        headers: {
          'x-error': `${owner}/${repo}/${ref}/fstab.yaml not found in bucket 'helix-code-bus'`,
        }
      });
```

File: src/index.js [hunk 4]
```diff
@@ -57,7 +59,33 @@ async function main(req, context) {
...
+  } finally {
+    /* istanbul ignore else */
+    if (codeStorage) {
+      codeStorage.close();
+    }
+  }
+
+  const mp = fstab.match(path);
   if (!mp) {
     return new Response(`path specified is not mounted in fstab.yaml: ${path}`, {
```
Comment (tripodsan) [inline src/index.js:90]:
```suggestion
    return new Response('', {
      headers: {
        'x-error': `path specified is not mounted in fstab.yaml: ${path}`,
      },  
```

File: src/storage.js [hunk 5]
```diff
@@ -138,19 +152,60 @@ class AWSStorage {
...
+      const result = await this.client.send(new GetObjectCommand(input));
+      log.info(`Object downloaded from: ${this.bucket}/${key}`);
+      const res = new Response(result.Body, {});
+      return await res.buffer();
+    } catch (e) {
+      /* istanbul ignore next */
+      if (e.$metadata.httpStatusCode !== 404) {
+        throw e;
+      }
+      return null;
```
Comment (tripodsan) [inline src/storage.js:190]:
```suggestion
      log.error(e);
      return null;
```

**Review by tripodsan — APPROVED** [review]

**Pull request merged**